examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/AuthenticationTest.kt

```diff
@@ -96,28 +88,62 @@ class AuthenticationTest: RealmTest() {
             // :remove-end:
             val user = app.login(Credentials.emailPassword(email, password))
             assertEquals(User.State.LOGGED_IN, user.state) // :remove:
+            user.delete() // :remove:
         }
         // :snippet-end:
+        app.close()
     }
 
     @Test
-    fun deleteUserTest() {
+    fun removeUserTest() {
         val email = getRandom()
         val password = getRandom()
-        // :snippet-start: user-delete
-        val app: App = App.create(YOUR_APP_ID) // Replace this with your App ID
-        runBlocking { // use runBlocking sparingly -- it can delay UI interactions
+        // :snippet-start: remove-user
+        val app = App.create(YOUR_APP_ID) // Replace with your App ID
+        runBlocking {
             // :remove-start:
             app.emailPasswordAuth.registerUser(email, password)
             val credentials = Credentials.emailPassword(email, password)
             // :remove-end:
+            // Log user in
             val user = app.login(credentials)
-            // use the user object ...
+            assertEquals(User.State.LOGGED_IN, user.state) // :remove:
+
+            // Work with logged-in user ...
+
+            // Remove the user from the app
+            // If the user is logged in, they are logged out first
+            // DOES NOT delete user from the server
+            user.remove()
+            // :remove-start:
+            assertEquals(User.State.REMOVED, user.state)
+            // :remove-end:
+        }
+        // :snippet-end:
+        app.close()
+    }
 
+    @Test
+    fun deleteUserTest() {
+        val email = getRandom()
+        val password = getRandom()
+        // :snippet-start: delete-user
+        val app: App = App.create(YOUR_APP_ID) // Replace with your App ID
+        runBlocking {
+            val credentials = Credentials.anonymous(reuseExisting = false) // :remove:
+            // Log user in
+            val user = app.login(credentials)
             assertEquals(User.State.LOGGED_IN, user.state) // :remove:
-            // later, delete the user object
-            user.delete() // regardless of which provider you used to login, you can logout using `delete()`
-            assertEquals(User.State.REMOVED, user.state) // :remove:
+
+            // Work with logged-in user ...
+
+            // Delete the logged-in user from the app
+            // and the Atlas App Services server
```

Comment:
Nit: we generally refer to "Atlas App Services App" - not server. We might want to clarify the wording here.
```suggestion
            // Delete the logged-in user from the device
            // and the Atlas App Services App
```

Reply:
updated